docs/HowToFAQ/Errors-Rewriter.md

```diff
@@ -0,0 +1,343 @@
+software is unaware of. So it is useful to always write constructors with parentheses, to visually
+distinguish them from simple identifiers.
+
+The `--warn-missing-constructor-parentheses` option will warn if a constructor is used without
+parentheses. The solution is to either add parentheses (if a constructor is intended) or
+rename the identifier (if a fresh bound identifier is intended).
+
+<!-- ./DafnyCore/RewritersPrecedenceLinter.cs-->
+
+## **Warning: unusual indentation in _what_ (which starts at _location_); do you perhaps need parentheses?**
+
+<!-- %check-resolve-warn -->
+```dafny
+function f(b: bool): bool
+{
+  b &&
+  b ==> b // this parses as (b &&b) ==> b, but reads as b && (b ==> b)
+}
+```
+
+Long expressions are best split up across multiple lines; readers often use indentation as a guide
+to the nesting of subexpressions. For example, parallel conjuncts or disjuncts may be written with 
+the same indentation. This warning occurs when the observed indentation is likely to cause
+a misunderstanding of the code, particularly of the precedence of operations. It is suggested
+to use parentheses or to redo the indentation to make the structure of the expression clearer.
+
+## **Warning: unusual indentation in _what_ (which ends at _location_); do you perhaps need parentheses?**
+
+<!-- %check-resolve-warn -->
+```dafny
+function f(b: bool): bool
+{
+  b ==> b && 
+  b          // this parses as b ==> (b && b), but reads as (b ==> b) && b
+}
+```
+
+Long expressions are best split up across multiple lines; readers often use indentation as a guide
+to the nesting of subexpressions. For example, parallel conjuncts or disjuncts may be written with 
+the same indentation. This warning occurs when the observed indentation is likely to cause
+a misunderstanding of the code, particularly of the precedence of operations. It is suggested
+to use parentheses or to redo the indentation to make the structure of the expression clearer.
+
+
+<!-- ./DafnyCore/Resolver/RunAllTestsMainMethod.cs -->
+
+## **Error: Cannot use /runAllTests on a program with a main method**
+
+<!-- TODO - this won't be triggered until Issue 3381 is fixed -->
+
+This error message when using `dafny test` and in the situation
+where a Main method is already present in the program.  
+`dafny test` synthesizes a Main method that calls all the test routines,
+which then conflicts with the existing Main method.
+(The `/runAllTests` option has been replaced by `dafny test`.)
+
+## **Error: Methods with the {:test} attribute can have at most one return value**
+
+<!-- %check-test -->
+```dafny
+method {:test} m(i: int) returns (j: int, k: int) 
+{
+  j, k := 0,0;
+}
+```
+
+This error only occurs when using `dafny test`. That command executes all methods attributed
+with `{:test}`, but such test methods are limited to methods with only one output parameter.
+This is purely an implementation limitation.
```

Comment:
If a user is trying to do this, can/should we recommend a workaround of returning a single tuple-typed value? Alternatively, if we plan to fix the implementation limitation, can we include a link to an issue so users can check its status?

Reply:
Made an issue, and a link here.